From e06b00e29126a0823322a74233a026dc29837273 Mon Sep 17 00:00:00 2001
From: Anatoliy Lysenko <a.o.lysenko@gmail.com>
Date: Sun, 27 Mar 2011 16:40:37 +0300
Subject: [PATCH] Added test for get has many through singular ids with conditions and include.

Test fail because of invalid sql:
ActiveRecord::StatementInvalid: SQLite3::SQLException: no such column: comments.id:
SELECT "posts".id FROM "posts" INNER JOIN "readers" ON "posts"."id" = "readers"."post_id"
WHERE "readers"."person_id" = 1 AND (comments.id is null)

Bug described in #6569 ticket.
---
 .../has_many_through_associations_test.rb          |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/activerecord/test/cases/associations/has_many_through_associations_test.rb b/activerecord/test/cases/associations/has_many_through_associations_test.rb
index 1efe342..e9bb5a8 100644
--- a/activerecord/test/cases/associations/has_many_through_associations_test.rb
+++ b/activerecord/test/cases/associations/has_many_through_associations_test.rb
@@ -714,6 +714,11 @@ class HasManyThroughAssociationsTest < ActiveRecord::TestCase
     assert_equal [categories(:general).id], authors(:mary).categories_like_general_ids
   end
 
+  def test_get_collection_singular_ids_on_has_many_through_with_conditions_and_include
+    person = Person.first
+    assert_equal person.posts_with_no_comment_ids, person.posts_with_no_comments.map(&:id)
+  end
+
   def test_count_has_many_through_with_named_scope
     assert_equal 2, authors(:mary).categories.count
     assert_equal 1, authors(:mary).categories.general.count
-- 
1.7.0.4

